### PR TITLE
WIP: Convert from using apischema to pydantic

### LINKED ISSUE #52: Use pydantic instead of apischema
As discussed

### PR DESCRIPTION
**Pull request by callumforrester**
Closes #52 

* Remove `apischema` as a dependency and use `pydantic` instead
* Use `pydantic.dataclasses.dataclass` instead of `dataclasses.dataclass`

### CODE REVIEW

File: src/scanspec/specs.py [hunk 1]
```diff
@@ -164,14 +157,12 @@ class Repeat(Spec[Axis]):
...
 
-    num: ANum
-    gap: A[
-        bool,
-        schema(
-            description="If False and the slowest of the stack of Frames is snaked "
-            "then the end and start of consecutive iterations of Spec will have no gap"
-        ),
-    ] = True
+    num: int = Field(min=1, description="Number of frames to produce")
```
Comment (coretl) [inline src/scanspec/specs.py:160]: Can we create Field instances up front and use them in two separate dataclasses?

Comment (callumforrester) [reply src/scanspec/specs.py:160]: Probably, but there have been issues with doing so in the past (https://github.com/pydantic/pydantic/issues/1412), so I wanted to get this working without first of all.

File: src/scanspec/regions.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
 from __future__ import annotations
 
 from dataclasses import dataclass, is_dataclass
+from email.policy import default
```
[the remark this answers is not in the document]
Comment (callumforrester) [reply src/scanspec/regions.py:4]: :upside_down_face:

File: src/scanspec/specs.py [hunk 3]
```diff
@@ -619,17 +599,17 @@ def calculate(self, bounds=True, nested=False) -> List[Frames[Axis]]:
             self._spiral_from_indexes, self.axes(), self.num, bounds
         )
 
-    @alternative_constructor
     def spaced(
```
[the remark this answers is not in the document]
Comment (callumforrester) [reply src/scanspec/specs.py:602]: I think so, but need to find a pydantic-friendly way to reintroduce them.

**Comment by callumforrester** [comment]
Superseded by https://github.com/dls-controls/scanspec/pull/56

**Pull request closed without merging**